database/src/postgres/mod.rs

```diff
@@ -0,0 +1,41 @@
+pub mod models;
+pub mod rpc_server;
+pub mod schema;
+pub mod state_indexer;
+pub mod tx_indexer;
+
+pub type PgAsyncPool =
+    diesel_async::pooled_connection::deadpool::Pool<diesel_async::AsyncPgConnection>;
+pub type PgAsyncConn =
+    diesel_async::pooled_connection::deadpool::Object<diesel_async::AsyncPgConnection>;
+
+#[async_trait::async_trait]
+pub trait PostgresStorageManager {
+    async fn create_pool(
+        database_url: &str,
+        database_user: Option<&str>,
+        database_password: Option<&str>,
+        database_options: crate::AdditionalDatabaseOptions,
+    ) -> anyhow::Result<PgAsyncPool> {
+        let connecting_url = if database_url.starts_with("postgres://") {
```

Comment:
`connection_string` would be a better name. I don't think the continuous tense is a good choice for variables like this.